src/Unreal.Core/BinaryReader.cs

```diff
@@ -23,7 +22,7 @@ public class BinaryReader : FArchive
         /// <seealso cref="System.IO.BinaryReader"/> 
         public BinaryReader(Stream input)
         {
-            using var ms = new MemoryStream();
+            using var ms = new MemoryStream((int) input.Length);
```

Comment:
Initializing the stream to the exact capacity we need reduces the time we need to spend resizing the data later.